.github/renovate.json5

```diff
@@ -0,0 +1,82 @@
+{
+  $schema: 'https://docs.renovatebot.com/renovate-schema.json',
+
+  baseBranches: ['next'],
+  enabledManagers: ['npm'],
+  extends: ['config:base', 'group:allNonMajor'],
+  fetchReleaseNotes: false, // Only fetch release notes for breaking changes
+  internalChecksFilter: 'strict', // PRs will be skipped unless a non-pending version is available (applies to "stabilityDays" check only)
+  labels: ['npm-dependencies'],
+  prConcurrentLimit: 1,
+  repositories: ['chanzuckerberg/edu-design-system'],
+  rangeStrategy: 'bump', // Always bump minor/patch to latest
+  reviewers: ['team:edu-design-system'],
+
+  major: {
+    rangeStrategy: 'replace',
+    fetchReleaseNotes: true,
+  },
+
+  ignoreDeps: [
+    // svg-spritemap-webpack-plugin v4 requires webpack 5 but EDS currently uses webpack 4 for storybook
+    'svg-spritemap-webpack-plugin',
+  ],
+
+  packageRules: [
+    {
+      // Hold back from creating branches until 3 or more days have elapsed since the version
+      // was released. Note: this only works for packages that provide a release timestamp header
+      // and does not ensure all updates are "stable".
+      matchDatasources: ['npm'],
+      stabilityDays: 3,
+    },
+    // Our peerDependencies for React are as low as 16.8.0, so we should support React v16
+    {
+      matchPackageNames: ['react', 'react-dom'],
+      groupName: 'react',
+      allowedVersions: '<17.0.0',
+    },
+    // Handle patch updates under 0.1.0 as potentially breaking and group them separately.
+    // Workaround for https://github.com/renovatebot/renovate/issues/3588
+    {
+      matchManagers: ['npm'],
+      matchCurrentVersion: '<0.1.0',
+      rangeStrategy: 'replace',
+      groupName: 'maybe breaking patch updates',
+      groupSlug: 'maybe-breaking-patch',
+      fetchReleaseNotes: true,
+    },
+    // Handle minor updates under 1.0.0 as potentially breaking and group them separately.
+    // Workaround for https://github.com/renovatebot/renovate/issues/3588
+    {
+      matchManagers: ['npm'],
+      matchCurrentVersion: '<1.0.0 >=0.1.0',
+      rangeStrategy: 'replace',
+      minor: {
+        groupName: 'maybe breaking minor updates',
+        groupSlug: 'maybe-breaking-minor',
+        fetchReleaseNotes: true,
+      },
+    },
+    // Updating packages with 'prettier' separately because updates to these packages often
+    // require modifying code to satistfy the new formatting rules.
+    {
+      matchPackagePatterns: ['prettier'],
+      groupName: 'prettier',
+    },
+    // Update typescript separately because typescript updates can create new
+    // typescript errors that need to be manually resolved.
+    {
+      matchPackageNames: ['typescript'],
+      groupName: 'typescript',
+    },
+    {
+      matchDepTypes: ['devDependencies'],
+      groupName: 'devDependencies',
+    },
+  ],
```

Comment:
I wonder if this is overriding the `group:allNonMajor` plugin and causing major updates to be grouped for devDependencies like in https://github.com/chanzuckerberg/edu-design-system/pull/1404

Reply:
Oh makes sense. How would you recommend then grouping `devDeps` separate from prod deps? Or does renovate do that automatically?

Reply:
Should I just land and see what happens on Monday?

Reply:
I'm not sure off the top of my head.

* What's the end state you're looking for?
* Generally, nonMajor updates are safe, what's the motivation for wanting to group devDeps and prodDeps separately and override the behavior of `group:allNonMajor`?

Reply:
That is a good question. The motivation was that through separation, if a bug rose from an update, the scope of reversion would be smaller. However I do agree that nonMajor updates are generally safe and this worry is probably trivial.